services/passport.js

```diff
@@ -11,14 +11,19 @@ passport.use(
         callbackURL: '/auth/google/callback',
         proxy: true
     }, async (accessToken, refreshToken, profile, done) => {
-        // Once the user has OAuth'ed with Google, we retrieve the access token and save them as a User model in our DB
-        var existingUser = await User.findOne({ googleId: profile.id });
 
-        if (existingUser) {
-            return done(null, existingUser); // Tells passport we're done w/ this user model
-        }
-        var user = new User({ googleId: profile.id }).save();
-        done(null, user);                    
+        dbService.getUserByGoogleID(profile.id, function(result) {
+            // If we've found the user via googleID, then she's already OAuthed.
+            if (result && result.length != 0 && result[0]) {
+                return done(null, result[0].ID);                
+            }
+            // If the user isn't OAuthed yet, insert the User model to our DB
+            else {
+                dbService.insertUser(profile, function(id) {
+                    done(null, id);
+                });
+            } 
+        });
```

Comment:
After the user OAuths, we'll see if they're an existing user.

If they are, we return the User model from the DB and serialize the ID into the cookie.

If they aren't, we insert the user and then return the ID for the cookie.